src/repository/storage/postgres/orm/sequelize/noteRelations.ts

```diff
@@ -209,4 +210,38 @@ export default class NoteRelationsSequelizeStorage {
 
     return foundNote !== null;
   };
+
+  /**
+   * Get all parent notes of a note that a user has access to,
+   * where the user has access to.
+   * @param noteId - the ID of the note.
+   */
+  public async getAllNoteParentsIds(noteId: NoteInternalId): Promise<NoteInternalId[]> {
+    const parentNotes: NoteInternalId[] = [];
+    let currentNoteId: number | null = noteId;
+
+    // get all note ids via a singe sql query instead of many
+    while (currentNoteId !== null) {
```

Comment:
have you tried to resolve it by a single sql query?

Reply:
I ll work on it